api/v1beta1/vaultdynamicsecret_types.go

```diff
@@ -61,6 +61,8 @@ type VaultDynamicSecretSpec struct {
 	RolloutRestartTargets []RolloutRestartTarget `json:"rolloutRestartTargets,omitempty"`
 	// Destination provides configuration necessary for syncing the Vault secret to Kubernetes.
 	Destination Destination `json:"destination"`
+	// SyncConfig configures sync behavior from Vault to VSO
+	SyncConfig *SyncConfig `json:"syncConfig,omitempty"`
```

Comment:
In the future we may want to have a separate DynamicSyncConfig type if we need options that aren't applicable to VaultStaticSecret's, but this is probably fine for now.